pkg/secrethub/auth.go

```diff
@@ -0,0 +1,28 @@
+package secrethub
+
+import "github.com/secrethub/secrethub-go/internals/auth"
+
+// AuthMethodService is an interface for any service that can provide authentication to the server.
+type AuthMethodService interface {
+	Authenticate() (auth.Authenticator, error)
+}
+
+// AuthService handles authentication to the SercretHub API.
+type AuthService interface {
+	AWS() AuthMethodService
```

Comment:
Why does this service have only one function? Should it also have `RSASigner` or `Session` or something like that? This suggests there is only one way to authenticate.

Reply:
If we add any other cloud services, we could add them here.

Reply:
Removed